dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/DotParse.java

```diff
@@ -1,159 +1,239 @@
-
-      final String contentLanguageId  = identifier + CONTENT_LANGUAGE_ID;
-      request.setAttribute(contentLanguageId, languageId);
-  }
-
-  private static long getContentLanguageId (final HttpServletRequest request, final String identifier,  final long defaultLanguageId) {
-
-      final String contentLanguageId  = identifier + CONTENT_LANGUAGE_ID;
-      return null != request &&
-              null != request.getAttribute(contentLanguageId)?
-                ConversionUtils.toLong(request.getAttribute(contentLanguageId), defaultLanguageId):
-                defaultLanguageId;
-  }
-
-
-  @Override
-  String resolveTemplatePath(final Context context, final Writer writer, final RenderParams params,final String[] arguments) {
-      
-    final String argument = arguments[0];
-    String templatePath   = argument;
-    Host host             = params.currentHost;
-    final User user       = params.user;
-    final HttpServletRequest request =
-            (HttpServletRequest) context.get("request");
-    
-    try {
-
-      // if we have a host
-      if (templatePath.startsWith(hostIndicator)) {
-        templatePath = templatePath.substring(hostIndicator.length(), templatePath.length());
-        String hostName = templatePath.substring(0, templatePath.indexOf('/'));
-        templatePath = templatePath.substring(templatePath.indexOf('/'), templatePath.length());
-        host = APILocator.getHostAPI().resolveHostName(hostName, user, params.mode.respectAnonPerms);
-      }
-
-      final Identifier identifier = APILocator.getIdentifierAPI().find(host, templatePath);
-      final long languageId = getContentLanguageId(request, identifier.getId(), params.language.getId());
-
-      //Verify if we found a resource with the given path
-      if ( null == identifier || !UtilMethods.isSet(identifier.getId()) ) {
-
-        String errorMessage = String.format("No resource found for [%s]",  arguments[0]);
+    public static void setContentLanguageId(final HttpServletRequest request, final String identifier, final long languageId) {
+
+        final String contentLanguageId = identifier + CONTENT_LANGUAGE_ID;
+        request.setAttribute(contentLanguageId, languageId);
+    }
+
+    private static long getContentLanguageId(final HttpServletRequest request, final String identifier,
+                    final long defaultLanguageId) {
+
+        final String contentLanguageId = identifier + CONTENT_LANGUAGE_ID;
+        return null != request && null != request.getAttribute(contentLanguageId)
+                        ? ConversionUtils.toLong(request.getAttribute(contentLanguageId), defaultLanguageId)
+                        : defaultLanguageId;
+    }
+
+
+    @Override
+    String resolveTemplatePath(final Context context, final Writer writer, final RenderParams params, final String[] arguments) {
+
+
+        String templatePath = arguments[0];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'templatePath' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5627236)